afs-ext-base/src/main/java/com/powsybl/afs/ext/base/ImportedCaseBuilder.java

```diff
@@ -134,14 +146,37 @@ public ImportedCase build() {
         if (context.getStorage().getChildNode(context.getFolderInfo().getId(), name).isPresent()) {
             throw new AfsException("Parent folder already contains a '" + name + "' node");
         }
-
-        // create project file
-        NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, ImportedCase.PSEUDO_CLASS, "", ImportedCase.VERSION,
-                new NodeGenericMetadata().setString(ImportedCase.FORMAT, importer.getFormat()));
-
-        // store case data
-        importer.copy(dataSource, new AppStorageDataSource(context.getStorage(), info.getId(), info.getName()));
-
+        NodeInfo info = null;
+        if (dataSource != null) {
+
+            // create project file
+            info = context.getStorage().createNode(context.getFolderInfo().getId(), name, ImportedCase.PSEUDO_CLASS, "", ImportedCase.VERSION,
+                    new NodeGenericMetadata().setString(ImportedCase.FORMAT, importer.getFormat()));
+            // store case data
+            importer.copy(dataSource, new AppStorageDataSource(context.getStorage(), info.getId(), info.getName()));
+        } else {
+            importer = Importers.findImporter(dataStore, name, importersLoader, context.getProject().getFileSystem().getData().getShortTimeExecutionComputationManager(), importConfig);
+            if (importer == null) {
+                throw new AfsException("No importer found for this data source");
+            }
+
+            // create project file
+            info = context.getStorage().createNode(context.getFolderInfo().getId(), name, ImportedCase.PSEUDO_CLASS, "", ImportedCase.VERSION,
+                    new NodeGenericMetadata().setString(ImportedCase.FORMAT, importer.getFormat()));
+            DataFormat df = importer.getDataFormat();
+            try {
+                Optional<DataPack> dp = df.newDataResolver().resolve(dataStore, this.name, this.parameters);
```

Comment:
Should not be needed anymore, but better use optional syntax for the sake of readability:
```java
DataPack dp = df.newDataResolver().resolve(dataStore, this.name, this.parameters)
                                                          .orElseThrow(() -> new AfsException("DataPack not resolved"));
dp.copyTo(...);
```